src/meshlink.c

```diff
@@ -2899,6 +2947,13 @@ ssize_t meshlink_channel_send(meshlink_handle_t *mesh, meshlink_channel_t *chann
 		return -1;
 	}
 
+	node_t *n = channel->node;
+
+	if(n->status.blacklisted) {
```

Comment:
I think this `if`-block should be removed. Since we already abort all UTCP connections when we blacklist a node, the call to `utcp_send()` later on will already return an error, and it's properly protected by the mesh mutex.